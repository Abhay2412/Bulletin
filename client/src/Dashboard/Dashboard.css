.Dashboard {
  text-align: center;
}

.Dashboard-backgound {
  background-color: white;
  min-height: 100vh;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: left;
}

.Profile-pic {
  position: absolute;
  width: 165px;
  height: 165px;
  top: 133px;
  left: calc(100vw / 2);
  border-radius: 75px;
}

.primary {
  position: absolute;
  width: 78px;
  height: 21px;
  left: calc((100vw / 2) + 50px);
  top: 308px;

  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen,
    Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 16px;

  color: #68390d;
}

.primary1 {
  position: absolute;
  width: 122px;
  height: 21px;
  left: calc((100vw / 2) + 25px);
  top: 329px;

  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen,
    Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 16px;

  color: #68390d;
}

.primary2 {
  position: absolute;
  width: 173px;
  height: 21px;
  left: calc(100vw / 2);
  top: 351px;

  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen,
    Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 16px;

  color: #68390d;
}

.primary3 {
  position: absolute;
  width: 122px;
  height: 21px;
  left: calc((100vw / 2) - 157px);
  top: 425px;

  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen,
    Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 16px;

  color: #68390d;
}

.primary4 {
  position: absolute;
  width: 122px;
  height: 21px;
  left: calc((100vw / 2) - 183px);
  top: 535px;

  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen,
    Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 16px;

  color: #68390d;
}

.primary5 {
  position: absolute;
  width: 122px;
  height: 21px;
  left: calc((100vw / 2) + 24px);
  top: 535px;

  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen,
    Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 16px;

  color: #68390d;
}

.primary6 {
  position: absolute;
  width: 122px;
  height: 21px;
  left: calc((100vw / 2) + 215px);
  top: 535px;

  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen,
    Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 16px;

  color: #68390d;
}

.input[type="text"] {
  position: absolute;
  width: 550px;
  height: 51px;
  left: calc((100vw / 2) - 140px);
  top: 455px;

  box-sizing: border-box;
  border: 0px;
  background: #cbdae1;
  outline: none;

  color: black;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen,
    Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  line-height: 16px;
}

.input[type="text"]:focus {
  border: 2px solid #ecd1c2;
}

.input1[type="text"] {
  position: absolute;
  width: 156px;
  height: 34px;
  left: calc((100vw / 2) - 140px);
  top: 565px;

  box-sizing: border-box;
  border: 0px;
  background: #cbdae1;
  outline: none;

  color: black;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen,
    Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  line-height: 16px;
}

.input1[type="text"]:focus {
  border: 2px solid #ecd1c2;
}

.input-date[type="date"] {
  position: absolute;
  width: 156px;
  height: 34px;
  left: calc((100vw / 2) + 60px);
  top: 565px;

  box-sizing: border-box;
  border: 0px;
  background: #cbdae1;
  outline: none;

  color: black;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen,
    Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  line-height: 16px;
}

.input-date[type="date"]:focus {
  border: 2px solid #ecd1c2;
}

.input-email[type="email"] {
  position: absolute;
  width: 149px;
  height: 34px;

  left: calc((100vw / 2) + 260px);
  top: 565px;

  box-sizing: border-box;
  border: 0px;
  background: #cbdae1;
  outline: none;

  color: black;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen,
    Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  line-height: 16px;
}

.input-email[type="email"]:focus {
  border: 2px solid #ecd1c2;
}

.side-panel {
  position: absolute;
  text-align: center;
  height: 100vh;
  width: 20vw;
  background: #cbdae1;
}

.logo {
  position: relative;
  width: 100px;
  height: 100px;
  top: 130px;
}

.nav {
  position: relative;
  top: 160px;

  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen,
    Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 16px;

  color: #68390d;
}

ul {
  margin: 1;
  padding: 0;
}

li a {
  display: block;
  color: #68390d;
  padding: 8px 16px;
  text-decoration: none;
}

li a.active {
  background-color: #ecd1c2;
  color: #68390d;
  font-weight: 400;
}

li a:hover {
  background-color: #ecd1c2;
  color: #68390d;
}