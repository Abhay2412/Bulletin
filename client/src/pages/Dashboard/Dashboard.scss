@import "../../styles/colors.scss";
.dashboard {
  width: 100%;
  height: 100%;

  margin-top: -120px;

  &-user-info {
    color: $color-brown;
    font-size: 1rem;
    &-container {
      height: 100%;
    }
  }

  &-grid-item {
    width: 100%;
    min-width: 200px;
    margin: auto;
  }

  &-grid-item > p {
    color: $color-brown;
  }

  .save-button {
    background-color: $color-pink;
    color: $color-brown;
    width: 30%;
    min-width: 150px;
    margin-bottom: 80px;
    margin-top: -15px;

    &-container {
      display: flex;
      align-items: center;
      justify-content: center;
    }
  }
}
