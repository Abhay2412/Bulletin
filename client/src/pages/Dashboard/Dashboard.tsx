import React from "react";
import profile from "./profile.svg";
import logo from "./logo.svg";
import "./Dashboard.css";

const Dashboard = () => {
    return (
        <div className="Dashboard">
            <header className="Dashboard-background">
                <div className="side-panel">
                    <img src={logo} className="logo"></img>
                    <ul className="nav">
                        <li><a className="active" href="#">My Profile</a></li>
                        <li><a href="#">Account Information</a></li>
                        <li><a href="#">Personalization</a></li>
                        <li><a href="#">Groups</a></li>
                        <li><a href="#">Support Us</a></li>
                    </ul>
                </div>

                <img src={profile} className="Profile-pic"></img>
                <h1 className="primary">John Doe</h1>
                <h1 className="primary1">Calgary, Alberta</h1>
                <h1 className="primary2">Last Login: 20/02/2022</h1>
                <h1 className="primary3">Brief Overview</h1>
                <h1 className="primary4">Name</h1>
                <h1 className="primary5">Birthday</h1>
                <h1 className="primary6">Email</h1>

                <form>
                    <input type="text" id="overview" placeholder="Tell everyone about yourself (max 150 characters)" className="input"></input>
                    <input type="text" id="name" placeholder="John Doe" className="input1"></input>
                    <input type="date" id="birthday" max="2012-12-31" className="input-date"></input>
                    <input type="email" id="email" placeholder="johndoe@gmail.com" className="input-email"></input>
                </form>
            </header>
        </div>
    );
};


export default Dashboard;