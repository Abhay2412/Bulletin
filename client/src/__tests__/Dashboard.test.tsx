import React from "react";
import Dashboard from "../pages/Dashboard/Dashboard";
import { configure, mount, ReactWrapper } from "enzyme";
import renderer from "react-test-renderer";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { store } from "../store";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

configure({ adapter: new Adapter() });

describe("Testing <Dashboard /> Component", () => {
  const wrapper: ReactWrapper = mount(
    <Provider store={store}>
      <BrowserRouter>
        <Dashboard />
      </BrowserRouter>
    </Provider>
  );

  it("should render properly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Dashboard />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should change overview field value", () => {
    wrapper
      .find("input")
      .find("#dashboard-overview-field")
      .simulate("change", {
        target: { value: "Test Overview....." },
      });
    expect(
      wrapper.find("input").find("#dashboard-overview-field").get(0).props.value
    ).toEqual("Test Overview.....");
  });

  it("should change name field value", () => {
    wrapper
      .find("input")
      .find("#dashboard-name-field")
      .simulate("change", {
        target: { value: "My Name" },
      });
    expect(
      wrapper.find("input").find("#dashboard-name-field").get(0).props.value
    ).toEqual("My Name");
  });
});
