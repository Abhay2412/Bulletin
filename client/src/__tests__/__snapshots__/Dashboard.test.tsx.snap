// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Dashboard /> Component should render properly 1`] = `
Array [
  <button
    className="MuiButtonBase-root MuiIconButton-root makeStyles-drawerButton-10"
    disabled={false}
    id="side-drawer-button"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
        data-testid="ArrowForwardIosIcon"
        focusable="false"
        style={
          Object {
            "color": "#68390D",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
        />
      </svg>
    </span>
  </button>,
  <div
    className="dashboard"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center"
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <div
        className="MuiGrid-root dashboard-user-info-container MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center"
      >
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <img
            className="dashboard-profile-pic"
            src="profile.svg"
          />
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <h1
            className="dashboard-user-info"
          />
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <h1
            className="dashboard-user-info"
          >
            Calgary, Alberta
          </h1>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <h1
            className="dashboard-user-info"
          >
            Last Login: 20/02/2022
          </h1>
        </div>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <div
            className="MuiGrid-root dashboard-grid-item MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-xl-10"
          >
            <p>
              Brief Overview
            </p>
            <div
              className="MuiFormControl-root MuiTextField-root makeStyles-textField-1"
              style={
                Object {
                  "width": "100%",
                }
              }
            >
              
              <div
                className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiFilledInput-input"
                  disabled={false}
                  id="dashboard-overview-field"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center MuiGrid-grid-xs-10 MuiGrid-grid-md-3 MuiGrid-grid-lg-3 MuiGrid-grid-xl-3"
          >
            <div
              className="MuiGrid-root MuiGrid-item"
              style={
                Object {
                  "width": "100%",
                }
              }
            >
              <p>
                Name
              </p>
              <div
                className="MuiFormControl-root MuiTextField-root makeStyles-textField-1"
              >
                
                <div
                  className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoFocus={false}
                    className="MuiInputBase-input MuiFilledInput-input"
                    disabled={false}
                    id="dashboard-name-field"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    type="text"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center MuiGrid-grid-xs-10 MuiGrid-grid-md-3 MuiGrid-grid-lg-3 MuiGrid-grid-xl-3"
          >
            <div
              className="MuiGrid-root MuiGrid-item"
              style={
                Object {
                  "width": "100%",
                }
              }
            >
              <p>
                Birthday
              </p>
              <div
                className="MuiFormControl-root MuiTextField-root makeStyles-textField-1"
                onClick={[Function]}
                onKeyPress={[Function]}
              >
                <div
                  className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl MuiInputBase-adornedEnd MuiFilledInput-adornedEnd"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoFocus={false}
                    className="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputAdornedEnd MuiFilledInput-inputAdornedEnd"
                    disabled={false}
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    placeholder="YYYY-MM-DD"
                    readOnly={false}
                    required={false}
                    type="text"
                    value="2022-03-28T16:28:47-06:00"
                  />
                  <div
                    className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-filled"
                  >
                    <button
                      className="MuiButtonBase-root MuiIconButton-root"
                      disabled={false}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <span
                        className="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                          />
                          <path
                            d="M0 0h24v24H0z"
                            fill="none"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                </div>
                
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center MuiGrid-grid-xs-10 MuiGrid-grid-md-3 MuiGrid-grid-lg-3 MuiGrid-grid-xl-3"
          >
            <div
              className="MuiGrid-root MuiGrid-item"
              style={
                Object {
                  "width": "100%",
                }
              }
            >
              <p>
                Email
              </p>
              <div
                className="MuiFormControl-root MuiTextField-root makeStyles-textField-1"
              >
                
                <div
                  className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline Mui-disabled Mui-disabled MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoFocus={false}
                    className="MuiInputBase-input MuiFilledInput-input Mui-disabled Mui-disabled"
                    disabled={true}
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    type="text"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="save-button-container"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained save-button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Save
        </span>
      </button>
    </div>
  </div>,
]
`;
